models/vyos/vyos_firewall/vyos_firewall.yaml

```diff
@@ -0,0 +1,212 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+        'metadata_version': '1.1',
+        'status': ['preview'],
+        'supported_by': 'network'
+    }
+
+NETWORK_OS: vyos
+RESOURCE: firewall_global
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: vyos_firewall_global
+  version_added: 2.10
+  short_description: Manage global policies or configurations for firewall on VyOS devices..
+  description: This moduleglobal policies or configurations for firewall on VyOS devices.
+  author:
+     - Rohit Thakur (@rohitthakur2590)
+  options:
+    config:
+      description: A dictionary of Firewall global configuration options.
+      type: dict
+      suboptions:
+        afi:
+          description:
+            - Specifies IP address type.
+          type: list
+          elements: dict
+          suboptions:
+            type:
+              description:  
+                - Specifies IP address type
+              type: str
+              choices: ['ipv4', 'ipv6']
+              required: True
+            source_route:
+              description:
+                - Specifies whether to process packets with the Loose Source Route or Strict Source Route IP options.
+                - (enable) Processes packets with source routing IP options set.
+                - (disable) Does not process packets with source routing IP options set.
+              type: dict
```

Comment:
take this with a grain of salt since I don't know the vyos firewall, but the description here with enable/disable defined seems odd to me, since this is a dict.  Rather, it seems like this detail belongs in the ip_src_rout parameter.